builtin/providers/github/resource_github_team_membership_test.go

```diff
@@ -0,0 +1,122 @@
+package github
+
+import (
+	"fmt"
+	"github.com/google/go-github/github"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"testing"
+)
+
+func TestAccGithubTeamMembership_basic(t *testing.T) {
+	var membership github.Membership
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubTeamMembershipDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGithubTeamMembershipConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGithubTeamMembershipExists("github_team_membership.test_team_membership", &membership),
+					testAccCheckGithubTeamMembershipRoleState("github_team_membership.test_team_membership", &membership),
+				),
+			},
```

Comment:
I think for this one we should have a 2nd `TestStep` that does an update to the role and we verify that it's recreated.